### PR TITLE
CREATEPROJECT, ENGINES: Add natvis for engines

### PR DESCRIPTION
**Pull request by henke37**
This makes it more user friendly to work with various engine types in the debugging panels.

### CODE REVIEW

File: devtools/create_project/scripts/scummvm-sludge.natvis [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+<AutoVisualizer xmlns="http://schemas.microsoft.com/vstudio/debugger/natvis/2010">
+	
+	<!-- Variables -->
+	
+	<Type Name="Sludge::Variable">
+		<DisplayString Condition="varType==SVT_INT">{varData.intValue}</DisplayString>
+		<DisplayString Condition="varType==SVT_FILE">File: {varData.intValue}</DisplayString>
+		<DisplayString Condition="varType==SVT_FUNC">Func: {Sludge::allUserFunc[varData.intValue]} (varData.intValue)</DisplayString>
+		<DisplayString Condition="varType==SVT_FUNC">Func: Unknown user function (varData.intValue)</DisplayString>
```
Comment (bluegr) [inline devtools/create_project/scripts/scummvm-sludge.natvis:17]: There are two identical conditions with different results here. Is this correct?

**Comment by bluegr** [comment]
Looks good at a first glance. What else needs to be done here?

File: devtools/create_project/scripts/scummvm-sludge.natvis (hunk 1 above)
Comment (henke37) [reply devtools/create_project/scripts/scummvm-sludge.natvis:17]: Good catch, but it's on purpose. Should the first one fail, the second will be used instead. Or so I interpret this.

**Comment by henke37** [comment]
I want to test the MTROPOLIS stuff, has been a pain to find a demo that doesn't `error()` instantly (something about a list modifier using an unsupported version).

And of course, I'd like to add support for more engines. And maybe even tweak `create_project` to automatically add the files to the project, removing the need to mess with the user global folder.

File: devtools/create_project/msbuild.cpp [hunk 2]
```diff
@@ -177,10 +177,15 @@ void MSBuildProvider::createProjectFile(const std::string &name, const std::stri
 	}
 
 	// Visual Studio 2015 and up automatically import natvis files that are part of the project
-	if (name == PROJECT_NAME && _version >= 14) {
-		project << "\t<ItemGroup>\n";
-		project << "\t\t<None Include=\"" << setup.srcDir << "/devtools/create_project/scripts/scummvm.natvis\" />\n";
-		project << "\t</ItemGroup>\n";
+	if (_version >= 14) {
+		std::string natvisFile = setup.srcDir + "/devtools/create_project/scripts/scummvm" + (name == PROJECT_NAME ? "" : "-"+name) +".natvis";
```
Comment (bluegr) [inline devtools/create_project/msbuild.cpp:181]:
```suggestion
		std::string natvisFile = setup.srcDir + "/devtools/create_project/scripts/scummvm" + (name == PROJECT_NAME ? "" : "-" + name) +".natvis";
```

**Comment by bluegr** [comment]
Looks good overall. Some small suggestions:
- Since we will potentially add more natvis files in the future, it would be good if they are in the same folder, e.g. `/devtools/create_project/natvis/`
- We no longer need [/devtools/create_project/scripts/install-natvis.bat](https://github.com/scummvm/scummvm/blob/master/devtools/create_project/scripts/install-natvis.bat), so it would be good if it was removed as part of this PR

**Comment by sev-** [comment]
This doesn't really belong to the main repository. If you want to maintain a set of these things, then a separate repo is more suitable.

**Comment by henke37** [comment]
This is just expanding an existing feature? Was it wrong to include it in the repo in the first place?

**Pull request closed without merging**